routes/items.js

```diff
@@ -18,14 +22,19 @@ router.post("/", (req, res, next) => {
 });
 
 router.put("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  if (currentItem(req.params.id)) {
+    res.status(200).send(itemData.updateById(req.params.id, req.body));
+  }
+  else (!currentItem(req.params.id))
+  res.status(200).send({ msg: `No item with the id of ${req.params.id}, creating item.` })
 });
```

Comment:
Above comments about reducing function calls and else conditionals apply here.

The logic of if the received id is valid (if `currentItem(req.params.id)` returns anything) should be left to the data interface method. Best practice is to keep the route handlers as ignorant as possible about data business logic.